File: RECURSION/JAVA/SubSequenceSum.java
package DSA.RECURSION.JAVA;
// print all subsequence sum = k
import java.util.*;

public class SubSequenceSum {
    public static void subsequencesum(int idx, int[] nums, List<Integer> sublist,int sum,int k){
        if(idx == nums.length){
            if(sum == k){
                System.out.print(sublist);
                System.out.println(": " + sum);
            }
            return;
        }
        sublist.add(nums[idx]);
        subsequencesum(idx+1, nums, new ArrayList<>(sublist), sum + nums[idx], k);

        sublist.remove(sublist.size() -1);
        subsequencesum(idx+1, nums, sublist, sum, k);
    }
    public static void main(String[] args) {
        int[] nums = {1,2,1};
        subsequencesum(0, nums, new ArrayList<>(), 0, 2);
    }
}

File: RECURSION/PYTHON/SubSequenceSum.py
#  print all subsequence sum = k

def SubSequenceSum(idx,nums,sublist, sumo, k):
    if idx == len(nums):
        if sumo == k:
            print(sublist, ": " , sumo)
        return

    SubSequenceSum(idx+1,nums,sublist + [nums[idx]],sumo + nums[idx], k)

    SubSequenceSum(idx+1,nums,sublist,sumo,k)

nums = [1,2,1]
SubSequenceSum(0,nums,[],0,2)